src/App.tsx

```diff
@@ -16,35 +18,94 @@
 ];
 
 export const App: React.FC = () => {
+  let goods = [...goodsFromServer];
+  let isReverseCLicked = false;
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortedGoods, setSortedGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function sortGoods() {
+    if (sortedGoods === ALPHABETICALLY) {
+      goods = goods.toSorted((goodA, goodB) => goodA.localeCompare(goodB));
+    }
+
+    if (sortedGoods === BY_LENGTH) {
+      goods = goods.toSorted((goodA, goodB) => goodA.length - goodB.length);
+    }
+
+    if (reverse) {
+      goods = goods.toReversed();
+    }
+  }
+
+  if (sortedGoods === RESET) {
+    goods = [...goodsFromServer];
+  }
+
+  if (isReverseCLicked) {
+    goods = goods.toReversed();
+    isReverseCLicked = false;
+  } else {
```

Comment:
[CODE KNOWLEDGE] - The same applies here. This condition should be inside the `sortGoods` function. Also, you should use the `setReversed` function to update the `reverse` state variable, not assign a new value directly.